src/cow-react/modules/limitOrders/hooks/useUpdateCurrencyAmount.ts

```diff
@@ -19,14 +22,19 @@ export function useUpdateCurrencyAmount() {
 
   return useCallback(
     (params: CurrencyAmountProps) => {
-      const update: CurrencyAmountProps = { ...params }
+      const update: Partial<Writeable<LimitOrdersState>> = {}
       const { inputCurrencyAmount, outputCurrencyAmount, keepOrderKind } = params
 
+      if (params.orderKind) {
+        update.orderKind = params.orderKind
+      }
+
       // Handle INPUT amount change
       if (inputCurrencyAmount !== undefined) {
         // Calculate OUTPUT amount by applying the rate
         const outputWithRate = applyLimitRate(inputCurrencyAmount, Field.INPUT)
-        update.outputCurrencyAmount = outputWithRate?.quotient.toString()
+        update.outputCurrencyAmount = fractionToString(outputWithRate)
+        update.inputCurrencyAmount = fractionToString(inputCurrencyAmount)
```

Comment:
No need to change in this PR. Just a open question to discuss about how do we deal with the global state.

I find it a bit strange how we save in the state a serialized version of the fraction.

Shouldn't our state use more complex types, instead of serilized version of them?

Reply:
Good question!
Since we persist state into localStorage, we have to use only primitive types.
For the deserialized values I added [useLimitOrdersTradeState](https://github.com/cowprotocol/cowswap/blob/develop/src/cow-react/modules/limitOrders/hooks/useLimitOrdersTradeState.ts#L35) hook.

Maybe would be better to use getter/setter of jotai atom for serialization/deserialisation, we should try it.